File: src/main/java/ru/hollowhorizon/hc/api/utils/IAnimatedEntity.java
package ru.hollowhorizon.hc.api.utils;

import ru.hollowhorizon.hc.client.render.entities.HollowAnimationManager;

public interface IAnimatedEntity {
    void processAnimation();

    void setAnimation(String animation);

    void setCustomAnimation(String animation);

    void endCustomAnimation();

    boolean isCustomAnimation();

    HollowAnimationManager getManager();
}

File: src/main/java/ru/hollowhorizon/hc/client/gltf/IAnimatedEntity.kt
package ru.hollowhorizon.hc.client.gltf

import ru.hollowhorizon.hc.client.gltf.animations.AnimationManager

interface IAnimatedEntity {
    var renderedGltfModel: RenderedGltfModel?
    var animationManager: AnimationManager?
}